Cubical/Categories/Constructions/Quotient.agda

```diff
@@ -0,0 +1,108 @@
+-- Quotient category
+{-# OPTIONS --safe #-}
+
+module Cubical.Categories.Constructions.Quotient where
+
+open import Cubical.Categories.Category.Base
+open import Cubical.Categories.Functor.Base
+open import Cubical.Categories.Limits.Terminal
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Prelude
+open import Cubical.HITs.SetQuotients renaming ([_] to ⟦_⟧)
+
+private
+  variable
+    ℓ ℓ' ℓq : Level
+
+module _ (C : Category ℓ ℓ') where
+  open Category C
+
+  module _ (_~_ : {x y : ob} (f g : Hom[ x , y ] ) → Type ℓq)
+           (~refl : {x y : ob} (f : Hom[ x , y ] ) → f ~ f)
+           (~cong : {x y z : ob}
+                    (f f' : Hom[ x , y ]) → f ~ f'
+                  → (g g' : Hom[ y , z ]) → g ~ g'
+                  → (f ⋆ g) ~ (f' ⋆ g')) where
+
+    private
+      Hom[_,_]/~ = λ (x y : ob) → Hom[ x , y ] / _~_
+
+    private module _ {x y z : ob} where
+      pre⋆/~ : Hom[ x , y ] → Hom[ y , z ]/~ → Hom[ x , z ]/~
+      pre⋆/~ f = rec squash/ (λ g → ⟦ f ⋆ g ⟧)
+                    λ g g' g~g' → eq/ _ _ (~cong _ _ (~refl _) _ _ g~g')
+
+      _⋆/~_ : Hom[ x , y ]/~ → Hom[ y , z ]/~ → Hom[ x , z ]/~
```

Comment:
Isn't there a `map2`or `rec2` function that you can use?

Reply:
Alternatively this: https://github.com/agda/cubical/blob/master/Cubical/HITs/SetQuotients/Properties.agda#L207

But I don't have time to think about it now

Reply:
Don't think so, it didn't work because the domain is two different quotients

Reply:
Right, but it should work after you generalize as you suggest in https://github.com/agda/cubical/issues/656 ?

Reply:
Should do yeah

Reply:
Ok, cool! Then please generalize and try. Write me when you're done and I'll take a look

Reply:
Redone using `rec2`